tests/integration/src/compiler_test.rs

```diff
@@ -584,37 +571,101 @@ impl CompilerTest {
-                    let mut name = LibraryPath::new(module_name).unwrap();
-                    name.set_namespace(miden_assembly::LibraryNamespace::Exec);
-                    let mut module = Module::new(name, ModuleKind::Executable);
-                    module.push_back(Box::new(entrypoint_function));
-                    module.imports.add(entrypoint);
-                    program.insert(Box::new(module));
-                }
-                program.into()
+                let user_ns_name = "user_ns";
+                let src =
+                    expected_masm_prog_source_from_modules(user_ns_name, &modules, self.entrypoint);
+                let prog = masm_prog_from_modules(user_ns_name, &modules, self.entrypoint);
+                (prog, src)
             }
         }
     }
 }
 
+// Assemble the VM MASM program from the compiled IR MASM modules
+fn masm_prog_from_modules(
+    user_ns_name: &str,
+    modules: &[Box<midenc_codegen_masm::Module>],
+    entrypoint: Option<FunctionIdent>,
+) -> Result<Program, Report> {
+    let mut assembler = Assembler::default().with_library(&StdLibrary::default())?;
+    for module in modules {
+        let module_src = format!("{}", module);
+        // eprintln!("{}", &module_src);
+        let path = masm_module_path(user_ns_name, module);
+        let library_path = LibraryPath::new(path).unwrap();
+        // dbg!(&library_path);
+        let options = miden_assembly::CompileOptions {
+            kind: ModuleKind::Library,
+            warnings_as_errors: false,
+            path: Some(library_path),
+        };
+        assembler.add_module_with_options(module_src, options)?;
+    }
+    if let Some(entrypoint) = entrypoint {
+        let prog_source = masm_prog_source(user_ns_name, entrypoint);
+        assembler.assemble(prog_source)
+    } else {
+        todo!()
+    }
+}
+
+// Generate the MASM program source code from the compiled IR MASM modules
+fn expected_masm_prog_source_from_modules(
+    user_ns_name: &str,
+    modules: &[Box<midenc_codegen_masm::Module>],
+    entrypoint: Option<FunctionIdent>,
+) -> String {
+    let mut src = String::new();
+    for module in modules {
+        let module_src = format!("{}", module);
+        let path = masm_module_path(user_ns_name, module);
+        if !path.contains("intrinsic") {
+            // print only user modules and not intrinsic modules
+            src.push_str(format!("# mod {path}\n\n").as_str());
+            src.push_str(&module_src);
+        }
+    }
+    if let Some(entrypoint) = entrypoint {
+        let prog_source = masm_prog_source(user_ns_name, entrypoint);
+        src.push_str(&prog_source);
+    } else {
+        todo!()
+    }
+    src
+}
+
+// Generate the MASM program source code (call the entrypoint function)
+fn masm_prog_source(user_ns_name: &str, entrypoint: FunctionIdent) -> String {
+    let module_name = entrypoint.module.as_str();
+    let function_name = entrypoint.function.as_str();
+    format!(
```

Comment:
Just as an aside, we won't need to explicitly import modules like this once 0xPolygonMiden/miden-vm#1363 is merged, we can switch to using fully-qualified procedure paths, e.g. `exec.::path::to::module::function` (note the leading `::`)